### DISCUSSION TITLE
Blacklist reaction speed

### DISCUSSION THREAD

**Discussion by flawrist** [DNSCrypt/dnscrypt-proxy#2249] [Q&A]
Dear @jedisct1,
even with a relatively small [blacklist](https://dblw.oisd.nl/basic/) (~1 MB) loaded in DNSCrypt 2.1.2 on Windows 7 x64,
I wait for a couple of seconds after executing `ping someBlockedAddress.tld` (e.g. `zykumw.com`)
before it reports `Ping request could not find host zykumw.com`,
whereas I expect it to report instantly.
No logging and other lists (e.g. `%systemroot%\system32\drivers\etc\hosts`) are used.

Why the delay?
Maybe something should be tuned in `dnscrypt-proxy.toml`?
Or `ping` is not a reliable way to measure the blacklist reaction speed?

**Comment by lifenjoiner** [comment]
Here are the differences:
```tsv
No.	Time	Source	Destination	Protocol	Length	Info
28	1.816854	127.0.0.1	127.0.0.1	DNS	60	Standard query 0x1879 A zykumw.com
29	1.817132	127.0.0.1	127.0.0.1	DNS	126	Standard query response 0x1879 A zykumw.com HINFO This query has been locally blocked by dnscrypt-proxy

65	3.958121	127.0.0.1	127.0.0.1	DNS	64	Standard query 0xd572 A ads.google.com
66	3.958240	127.0.0.1	127.0.0.1	DNS	80	Standard query response 0xd572 A ads.google.com A 0.0.0.0

22	1.672133	127.0.0.1	127.0.0.1	DNS	59	Standard query 0x85fa A 003999.cc
23	1.672317	127.0.0.1	127.0.0.1	DNS	59	Standard query response 0x85fa A 003999.cc
```
It returns a `HINFO` rather than an IP nor an empty Answer.

That seems neither to be recognized by `ping` nor IE, on Win7; but, it is OK on Win10 22H2.

**Reply by flawrist** [reply]
I changed `blocked_query_response` from `hinfo` to `refused` in `dnscrypt-proxy.toml`, but it did not help, However, `a:0.0.0.0` did, the delay vanished. Should I continue to use this option? Cons?

**Reply by lifenjoiner** [reply]
I think you've gotten the answer by yourself.

**Reply by flawrist** [reply]
@lifenjoiner, I stumbled upon the solution by accident, by playing with settings. After which I asked if this solution had any cons. Because the default setting is `hinfo` for some reason.

**Reply by lifenjoiner** [reply]
https://github.com/DNSCrypt/dnscrypt-proxy/blob/5705b609368d73a216efba710dfaac7642a7b3a9/dnscrypt-proxy/example-dnscrypt-proxy.toml#L141-L146
The comments have explained, and you say it works. Why not?
My local server always return IP/empty answers for blocked domains, on Windows. Cons are welcome to be reported.
